### PR TITLE
Web crawling for cnbc 1st PR

### PR DESCRIPTION
**Pull request by Dazeofcitywalk**
Summary:
Create the `get_news_from_cnbc` function to crawl articles from http://www.cnbc.com

Details:
· detect link with "PRO" or "CLUB" icon and avoid crawling automatically
· detect video link and avoid crawling automatically
· crawl the accessible articles and store them to txt file

Screenshots:
Screen Shot 2023-08-01 at 6 07 42 PM
Take it as example. Only the second article can be crawled, and it is stored in the txt file.

### CODE REVIEW

File: Phraends_Flask/Backend/Crawler/Crawler.py [hunk 1]
```diff
@@ -178,3 +178,69 @@ def get_news_from_dow_jones(ticker):  # apple
...
+            # window_handles[0] is a first window
+            driver.switch_to.window(driver.window_handles[0])
+            skip_sign = 1
+        except:
+            continue
+
+    driver.quit()
+    return (
+        links,
+        open("the_news_texts.txt", mode="r"),
```
Comment (alex308248) [inline Phraends_Flask/Backend/Crawler/Crawler.py:243]: As we discussed last week, you do not need to write data into a txt file.

File: Phraends_Flask/Backend/Crawler/Crawler.py [hunk 2]
```diff
@@ -178,3 +178,69 @@ def get_news_from_dow_jones(ticker):  # apple
...
+        By.XPATH, '/html/body/div[3]/div/div[1]/header/div[4]/div/div/div[3]/section/div[2]/div[1]/section/form/input[2]'
+    )
+    search.send_keys(str(ticker))
+    search.send_keys(Keys.RETURN)
+
+    # Collect the 5 latest news content
+    links = []
+    driver.implicitly_wait(5)
+    f = open("the_news_texts.txt", "w")
+    for i in range(0, 5):
```
Comment (alex308248) [inline Phraends_Flask/Backend/Crawler/Crawler.py:223]: Could you modified it and make it returning 5 latest articles that is availabe? I believe using while loop will be an easy implementation.

**Review by alex308248 — CHANGES_REQUESTED** [review]
You'll need to delete the txt file and make the function return 5 articles as a list. As we discussed last week, we will longer use txt file to read and write artilces through crawler and model api. In addition, remeber to pull the lastest version of main branch and fix the conflicts.

**Comment by Dazeofcitywalk** [comment]
Thanks for reviewing.
For article extension, this website only provides 5 latest articles on it, so I couldn't crawl more articles. 
For txt-list modification, I'm currently debugging, after that I will change the style of return.

**Comment by alex308248** [comment]
> Thanks for reviewing. For article extension, this website only provides 5 latest articles on it, so I couldn't crawl more articles. For txt-list modification, I'm currently debugging, after that I will change the style of return.

OK~

**Comment by YLee2121** [comment]
have merge conflict, please solve it

**Comment by Dazeofcitywalk** [comment]
Here is my update:
1. change return style from txt to list
2. test more tickers, and add some try except blocks based on it
3. resolved the merge conflict

Thank you!

**Review by alex308248 — APPROVED** [review]
Approved~

**Pull request merged**